### PR TITLE
Propagate the labels form the build on the pod

### PR DESCRIPTION
**Pull request by ccojocar**
This allows usrs to specify custom labels.

### CODE REVIEW

**Comment by ccojocar** [comment]
@ImJasonH Any chance to have a look over this PR? Thanks

File: pkg/reconciler/build/resources/pod.go [hunk 1]
```diff
@@ -259,6 +259,14 @@ func MakePod(build *v1alpha1.Build, kubeclient kubernetes.Interface) (*corev1.Po
 	}
 	annotations["sidecar.istio.io/inject"] = "false"
 
+	// Copy labels on the build through to the underlying pod to allow users
+	// to specify pod lablels.
```
Comment (jonjohnsonjr) [inline pkg/reconciler/build/resources/pod.go]: typo: labels

**Comment by ccojocar** [comment]
@vdemeester The tests passed after rebasing. Please could have a look if it can be merged? Thanks

**Pull request merged**